lizrd/support/test_misc.py

```diff
@@ -48,46 +81,98 @@ def test_multiple_transitions(self):
         ]
 
         for processed_tokens, expected_batch_size in test_cases:
-            actual_batch_size = calculate_current_batch_size_from_rampup(
-                processed_tokens,
-                transition_points,
-                batch_sizes,
-                target_batch_size,
+            step = convert_tokens_to_steps(
+                tokens=processed_tokens,
+                seq_len=seq_len,
+                target_batch_size=target_batch_size,
+                transition_points=transition_points,
+                batch_sizes=batch_sizes,
+            )
+            actual_batch_size = get_batch_size(
+                step=step,
+                target_batch_size=target_batch_size,
+                transition_points=config.transition_points,
+                batch_sizes=config.batch_sizes,
             )
             self.assertEqual(actual_batch_size, expected_batch_size)
 
+    def test_single_transition(self):
+        # Test with a single transition point
+        transition_points = [0.001, 0.01, 0.1, 1.0]  # in billions
+        batch_sizes = [1e2, 1e3, 1e4, 1e5]
+        seq_len = 1e3
+
+        actual_transition_points = convert_transition_points_in_tokens_to_steps(
+            transition_points_in_tokens=transition_points,
+            batch_sizes=batch_sizes,
+            seq_len=seq_len,
+        )
+
+        # for example we need 10 steps with bs 1e4 and seq_len 1e3 to get 1e8 tokens, but one 10th of the tokens was completed before we switched batch size
+        expected_transition_points = [10, 19, 28, 37]
+        self.assertEqual(actual_transition_points, expected_transition_points)
```

Comment:
```suggestion
class ConvertTransitionPointsInTokensToStepsTest
    def test_single_transition(self):
        # Test with a single transition point
        transition_points = [0.001, 0.01, 0.1, 1.0]  # in billions
        batch_sizes = [1e2, 1e3, 1e4, 1e5]
        seq_len = 1e3

        actual_transition_points = convert_transition_points_in_tokens_to_steps(
            transition_points_in_tokens=transition_points,
            batch_sizes=batch_sizes,
            seq_len=seq_len,
        )

        # for example we need 10 steps with bs 1e4 and seq_len 1e3 to get 1e8 tokens, but one 10th of the tokens was completed before we switched batch size
        expected_transition_points = [10, 19, 28, 37]
        self.assertEqual(actual_transition_points, expected_transition_points)
```

Reply:
This is a test for a separate function. I think it makes more sense to have a separate test class for it. Otherwise, we have two class methods with the same name in `CalculateCurrentBatchSizeTest` class